components/validator/guardian.go

```diff
@@ -34,25 +34,39 @@ type Guardian struct {
 	cancel context.CancelFunc
 	wg     sync.WaitGroup
 
+	submissionInterval *big.Int
+
+	l2ooContract            *bindings.L2OutputOracle
```

Comment:
I didn't add it to `guardian`, because it creates new context with timeout every network request. Otherwise, we can use newly created context with timeout also in `challenger`. Would it be better to use it also in `challenger`?

Reply:
I see. I think it's out of scope of this PR. Let's keep this way for now.
It seems better to add timeout as you said.